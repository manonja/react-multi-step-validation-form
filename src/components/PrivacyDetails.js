import React, { Component } from 'react'
export class PrivacyDetails extends Component {
  // Go to the next page on submit
  continue = e => {
    e.preventDefault()
    this.props.nextStep()
  }

  // Go back to the previous page 
  previous = e => {
    e.preventDefault()
    this.props.prevStep()
  }

  render() {
    const {handlePrivacy, comTrayProduct, comOtherProducts} = this.props

    return (
      <div className='wrapper'>
        <div className='form-wrapper'>
        <div className='page-detail'>
          <p className="alignleft user">User Details</p>
          <h3 className="aligncenter privacy">Privacy</h3>
          <p className="alignright done">Done</p>
        </div>
        <div className='privacy-options'>
          <div>
            <input 
              type="checkbox" 
              value="comTrayProduct" 
              checked={comTrayProduct} 
              onChange={handlePrivacy('comTrayProduct')} 
            />
            <label 
              id='label-txt' 
              htmlFor="comTrayProduct">Receive updates about Tray.io product by email
            </label>
          </div>
          <br/>
          <div>
            <input 
              type="checkbox" 
              value="comOtherProducts" 
              checked={comOtherProducts} 
              onChange={handlePrivacy('comOtherProducts')}
            />            
            <label 
              id='label-txt' 
              htmlFor="comOtherProducts">Receive communication by email for other products created by the Tray.io team
            </label>
          </div> 
          <br/> 
        </div>
       
        <br/>
        <button id='btn-next' onClick={this.continue}>Next</button>
        <br/>
        <button id='btn-back' onClick={this.previous}>Back</button>
   
        </div>    
    </div>         
      
    )
  }
}



export default PrivacyDetails
