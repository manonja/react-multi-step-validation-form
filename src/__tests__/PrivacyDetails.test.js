import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import PrivacyDetails from '../components/PrivacyDetails';


let wrapper;

beforeEach(() => {
  wrapper = shallow(
    <PrivacyDetails 
      handlePrivacy={({}) => {}} 
      comTrayProducts=''
      comOtherProducts=''
      nextStep={() => {}} 
      prevStep={() => {}}
    />
  )
});

describe('<PrivacyDetails /> rendering', () => {
  it('PrivacyDetails: renders correctly', () => {
    const tree = renderer.create(
        <PrivacyDetails 
          handlePrivacy={({}) => {}} 
          comTrayProducts=''
          comOtherProducts=''
          nextStep={() => {}} 
          prevStep={() => {}}
        />).toJSON() 
    
    expect(tree).toMatchSnapshot();
});

  it('renders without crashing', () => {
    expect(wrapper)
  });

  it('should render 2 inputs fields for checkboxes', () => {
    expect(wrapper.find('input').length).toEqual(2);      
  });

  it('should render 2 buttons', () => {
    expect(wrapper.find('button').length).toEqual(2);      
  });
});

