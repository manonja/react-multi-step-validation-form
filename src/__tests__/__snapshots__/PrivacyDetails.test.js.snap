// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivacyDetails /> rendering PrivacyDetails: renders correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="form-wrapper"
  >
    <div
      className="page-detail"
    >
      <p
        className="alignleft user"
      >
        User Details
      </p>
      <h3
        className="aligncenter privacy"
      >
        Privacy
      </h3>
      <p
        className="alignright done"
      >
        Done
      </p>
    </div>
    <div
      className="privacy-options"
    >
      <div>
        <input
          type="checkbox"
          value="comTrayProduct"
        />
        <label
          htmlFor="comTrayProduct"
          id="label-txt"
        >
          Receive updates about Tray.io product by email
        </label>
      </div>
      <br />
      <div>
        <input
          checked=""
          type="checkbox"
          value="comOtherProducts"
        />
        <label
          htmlFor="comOtherProducts"
          id="label-txt"
        >
          Receive communication by email for other products created by the Tray.io team
        </label>
      </div>
      <br />
    </div>
    <br />
    <button
      id="btn-next"
      onClick={[Function]}
    >
      Next
    </button>
    <br />
    <button
      id="btn-back"
      onClick={[Function]}
    >
      Back
    </button>
  </div>
</div>
`;
